syndeo-template/src/main/java/com/google/cloud/syndeo/transforms/datagenerator/DataGeneratorSchemaTransform.java

```diff
@@ -0,0 +1,103 @@
+import org.apache.beam.sdk.schemas.utils.AvroUtils;
+import org.apache.beam.sdk.transforms.FlatMapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.PeriodicImpulse;
+import org.apache.beam.sdk.transforms.Reshuffle;
+import org.apache.beam.sdk.transforms.SimpleFunction;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionRowTuple;
+import org.apache.beam.sdk.values.Row;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+
+public class DataGeneratorSchemaTransform implements SchemaTransform, Serializable {
+
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private long recordsPerSecond = 1000l;
+  private long minutesToRun = 10;
+  private String schema;
+
+  public DataGeneratorSchemaTransform(long recordsPerSecond, long minutesToRun, String schema) {
+    this.recordsPerSecond = recordsPerSecond;
+    this.minutesToRun = minutesToRun;
+    this.schema = schema;
+  }
+
+  @Override
+  public PTransform<PCollectionRowTuple, PCollectionRowTuple> buildTransform() {
+    return new PTransform<PCollectionRowTuple, PCollectionRowTuple>() {
+      @Override
+      public PCollectionRowTuple expand(PCollectionRowTuple input) {
+        org.apache.beam.sdk.schemas.Schema beamSchema =
+            AvroUtils.toBeamSchema(new Parser().parse(schema));
+        final Instant startTime = Instant.now();
+        PCollection<Instant> instants =
+            input
+                .getPipeline()
+                .apply(
+                    PeriodicImpulse.create()
+                        .startAt(startTime)
+                        .stopAt(Instant.now().plus(Duration.standardMinutes(minutesToRun)))
+                        .withInterval(Duration.millis(1000))
+                        .applyWindowing());
+        instants = instants.apply(Reshuffle.viaRandomKey());
+
+        PCollection rows =
+            instants
+                .apply(FlatMapElements.via(new InstantToRowFn(startTime, schema)))
+                .setRowSchema(beamSchema);
+        return PCollectionRowTuple.of("output", rows);
+      }
+    };
+  }
+
+  private class InstantToRowFn extends SimpleFunction<Instant, List<Row>> {
+
+    private Instant startTime;
+    private String schema;
+    private String format;
+
+    public InstantToRowFn(Instant startTime, String schema) {
+      this.startTime = startTime;
+      this.schema = schema;
+    }
+
+    @Override
+    public List<Row> apply(Instant input) {
+      final Schema schema = Schema.parse(this.schema);
+
+      final long numSplits = recordsPerSecond * 60;
+      final long periodPerSplitMsecs = 1000;
+      long ordinal =
+          (input.minus(Duration.millis(startTime.getMillis())).getMillis() / periodPerSplitMsecs);
+      return LongStream.range(ordinal * recordsPerSecond, (ordinal + 1) * recordsPerSecond)
+          .mapToObj(intVal -> RecordCreator.createRowRecord(schema))
```

Comment:
if we parse the schema into a Beam schema early, then we don't need to parse it into avro-and-then-into-beam-schema in each call here (createRowRecord)